samples/python/evaluate_qrcode/evaluate_qrcode.py

```diff
@@ -0,0 +1,386 @@
+
+    if metric <= 3:
+        print_result(categories, detection_stat, decoding_stat)
+    else:
+        print_result_scores(categories, scores)
+
+
+def get_input_images(path_to_images):
+    input_images = [f for f in sorted(os.listdir(path_to_images)) if f.endswith("jpg") or f.endswith("JPG") or f.endswith("png")]
+
+    return input_images
+
+
+def get_input_data(path_to_data, input_file):
+    path_to_file = os.path.join(path_to_data, input_file)
+    json_file = open(path_to_file, 'r')
+    input_data = json.load(json_file)
+
+    return input_data
+
+
+def print_metric_description(metric):
+    if metric == 1:
+        print("The percentage of pictures in each category where all QR-codes on the picture are detected/decoded\n")
+    elif metric == 2:
+        print("The percentage of pictures in each category where at least one QR-code on the picture is detected/decoded\n")
+    elif metric == 3:
+        print("The percentage of QR-codes detected/decoded among all QR-codes in each category\n")
+    elif metric == 4:
+        print("Intersection over Union\n")
+    elif metric == 5:
+        print("F1-score\n")
+
+
+def get_number_qrcodes_in_category(path_to_data, input_file):
+    number_qrcodes = 0
+
+    input_data = get_input_data(path_to_data, input_file)
+    total_number_images = len(input_data['test_images'])
+
+    for idx in range(total_number_images):
+        image_data = input_data['test_images'][idx]
+        number_qrcodes += len(image_data['points'])
+
+    return number_qrcodes
+
+
+def print_result(categories, detection_stat, decoding_stat):
+    for idx, category in enumerate(categories):
+        print("Category {}".format(category))
+        print("detected percent: {} %".format(np.round(detection_stat[idx] * 100, 2)))
+        print("decoded percent: {} %\n".format(np.round(decoding_stat[idx] * 100, 2)))
+
+
+def print_result_scores(categories, scores):
+    for idx, category in enumerate(categories):
+        print("Category {}".format(category))
+        print("Score: {} %\n".format(np.round(scores[idx] * 100, 2)))
+
+
+def compute_metric_1(image_data, image_to_detect, qrDetector):
+    detected = 0
+    decoded = 0
+
+    expected_number_qrcodes = len(image_data['points'])
+
+    ok, bboxes = qrDetector.detectMulti(image_to_detect)
+
+    if bboxes is not None:
+        if expected_number_qrcodes == len(bboxes):
+            detected += 1
+
+        ok, decoded_data, rec = qrDetector.decodeMulti(image_to_detect, bboxes)
+
+        if len(decoded_data) > 0:
+            number_decoded = 0
+            for data in decoded_data:
+                if len(data) > 0:
+                    number_decoded += 1
+            if number_decoded == expected_number_qrcodes:
+                decoded +=1
+
+    return detected, decoded
+
+
+def compute_metric_2(image_to_detect, qrDetector):
+    detected = 0
+    decoded = 0
+
+    ok, bboxes = qrDetector.detectMulti(image_to_detect)
```

Comment:
what does `ok` variable mean?

Reply:
quirc is updated as well, current qr code decoder is buggy especially using video capture.

Reply:
> quirc is updated as well, current qr code decoder is buggy especially using video capture.

thank you, we'll try update quirc